doc/vxlan/Overlay ECMP with BFD.md

```diff
@@ -0,0 +1,248 @@
+
+#### Pros of SWSS to handle route update based on tunnel nexthop health: 
+
+- No significant changes, if BFD session management is HW offload via SAI notifications or Control Plane assisted. 
+- Similar to NHFLAGS handling for existing route ECMP group 
+- Better performance in re-programming routes in ASIC instead of separate process to monitor and modify each route prefix by updating DB entries 
+
+### BfdOrch
+Sonic shall offload the BFD session handling to hardware that has BFD capabilities.  A new module, BfdOrch shall be introduced to handle BFD session to monitoring endpoints and check the health of remote endpoints. BfdOrch shall offload the session initiation/sustenance to hardware via SAI APIs and gets the notifications of session state from SAI. The session state shall be updated in STATE_DB and to any other observer orchestration agents.  
+
+![](https://github.com/Azure/SONiC/blob/master/images/vxlan_hld/OverlayEcmp_BFD.png)
+
+For offloading, the following shall be the SAI attributes programmed by BfdOrch.  
+
+|  Attribute Type          |  Value                         |
+|--------------------------|--------------------------------|
+| SAI_BFD_SESSION_ATTR_TYPE | SAI_BFD_SESSION_TYPE_ASYNC_ACTIVE  |
+| SAI_BFD_SESSION_ATTR_OFFLOAD_TYPE | SAI_BFD_SESSION_OFFLOAD_TYPE_FULL |
+| SAI_BFD_SESSION_ATTR_BFD_ENCAPSULATION_TYPE | SAI_BFD_ENCAPSULATION_TYPE_NONE |
+| SAI_BFD_SESSION_ATTR_SRC_IP_ADDRESS | Loopback0 IPv4 or v6 address |
+| SAI_BFD_SESSION_ATTR_DST_IP_ADDRESS | Remote IPv4 or v6 address |
+| SAI_BFD_SESSION_ATTR_MULTIHOP | True |
+
+Sai shall notify via notification channel on the session state as one of sai_bfd_session_state_t. BfdOrch can listen on these notifications and update the StateDB for the session state.
+
+The flow of BfdOrch is presented in the following figure. BfdOrch subscribes to the BFD_SESSION_TABLE of APPL_DB and send the corresponding request to program the BFD sessions to syncd accordingly. The BfdOrch also creates the STATE_DB entry of the BFD session which includes the BFD parameters and an initial state. Upon receiving bfd session state change notifications from syncd, BfdOrch update the STATE_DB field to update the BFD session state. 
+
+![](https://github.com/Azure/SONiC/blob/master/images/vxlan_hld/OverlayEcmp_BFD_Notification.png)
+
+A Control Plane BFD approach is to use FRR BFD and enable bfdctl module. This shall be part of the Sonic BGP container. For the current usecase, the BFD hw offload is being considered and control plane BFD using FRR is not scoped in this document.
```

Comment:
With HW assisted BFD, can we have control plane BFD also functioning? When BFD packets are received, what will be the criteria to differentiae between HW assisted BFD packets and control plane handled BFD packets? please document..

Reply:
Added to https://github.com/Azure/SONiC/pull/880